airbyte-commons-worker/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java

```diff
@@ -117,18 +119,20 @@ public DefaultReplicationWorker(final String jobId,
                                   final RecordSchemaValidator recordSchemaValidator,
                                   final WorkerMetricReporter metricReporter,
                                   final ConnectorConfigUpdater connectorConfigUpdater,
-                                  final boolean fieldSelectionEnabled) {
+                                  final boolean fieldSelectionEnabled,
+                                  final HeartbeatTimeoutChaperone srcHeartbeatTimeoutChaperone) {
     this.jobId = jobId;
     this.attempt = attempt;
     this.source = source;
     this.mapper = mapper;
     this.destination = destination;
     this.messageTracker = messageTracker;
-    this.executors = Executors.newFixedThreadPool(2);
+    this.executors = Executors.newFixedThreadPool(4);
```

Comment:
Should we add a dedicated a threadpool in the HeartbeatTimeoutChaperone? Given that in this context we're looking at fixed expected number of long running threads, feels like injection the threadpool is more error prone.

Reply:
Yes, that makes senses, done.